tasks/ssh_private_key.yaml

```diff
@@ -0,0 +1,18 @@
+- name: ensure ssh private key is set
+  loop: |
+    {%- set private_keys = [] -%}
+    {%- for user_name, user_value in system.users.items() -%}
+      {%- if user_value.private_key is defined -%}
+        {%- set _ = private_keys.append({'user': user_name, 'key': user_value.private_key}) -%}
+      {%- endif -%}
+    {%- endfor -%}
+    {{ private_keys }}
+  loop_control:
+    label: "{{ item.user }}"
+  copy:
+    dest: "~{{ item.user }}/.ssh/id_ed25519"
```

Comment:
`id_ed25519` should not be hardcoded. This should either be an option or automatically detect the key type.